pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -132,6 +133,12 @@ func (h *upgradeHandler) OnChanged(key string, upgrade *harvesterv1.Upgrade) (*h
 		return h.upgradeClient.Update(toUpdate)
 	}
 
+	// only run further operations for latest upgrade
+	if upgrade.Labels == nil || upgrade.Labels[harvesterLatestUpgradeLabel] != "true" {
```

Comment:
prefer to add a log here with proper message and label info.

Reply:
Added it. Thank you.